components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java

```diff
@@ -487,117 +337,70 @@ private String encrypt(String plainText) throws CryptoException {
         }
     }
 
     /**
-     * Validate that secret type is non-empty.
-     * Validate the secret type is exist
-     *
-     * @param secretTypeName The secret type name to be deleted.
-     * @throws SecretManagementException If secret validation fails.
+     * Validate secret value against a regex pattern.
+     * @param secretValue Value of the secret.
+     * @throws SecretManagementClientException
      */
-    private void validateSecretTypeDeleteRequest(String secretTypeName) throws SecretManagementException {
-
-        if (StringUtils.isEmpty(secretTypeName)) {
-            if (log.isDebugEnabled()) {
-                log.debug("Invalid secret type name: " + secretTypeName + ".");
-            }
-            throw handleClientException(ERROR_CODE_SECRET_TYPE_NAME_REQUIRED, secretTypeName, null);
-        }
-
-        if (!isSecretTypeExists(secretTypeName)) {
-            if (log.isDebugEnabled()) {
-                log.debug("A secret type with the name: " + secretTypeName + " does not exists.");
-            }
-            throw handleClientException(ERROR_CODE_SECRET_TYPE_DOES_NOT_EXISTS, secretTypeName);
-        }
-    }
-
-    private String retrieveOrGenerateSecretTypeId(String secretTypeName) throws SecretManagementException {
-
-        String secretTypeID;
-        if (isSecretTypeExists(secretTypeName)) {
-            secretTypeID = getSecretType(secretTypeName).getId();
-        } else {
-            secretTypeID = generateUniqueID();
+    private void validateSecretValue(String secretValue) throws SecretManagementClientException {
+        if (!isSecretValueRegexValid(secretValue)) {
             if (log.isDebugEnabled()) {
-                log.debug("Secret type id generated: " + secretTypeID);
+                log.debug("Secret value does not conform to "
+                        + getSecretValueRegex() + " pattern");
             }
+            throw handleClientException(ErrorMessages.ERROR_CODE_INVALID_SECRET_VALUE, getSecretValueRegex());
         }
-        return secretTypeID;
     }
 
     /**
-     * Validate that secret type is non-empty.
-     * Validate the secret type is exist
-     *
-     * @param secretType The secret type to be updated.
-     * @throws SecretManagementException If secret validation fails.
+     * Validate secret description against a regex pattern.
+     * @param description Description of the secret.
+     * @throws SecretManagementClientException
      */
-    private void validateSecretTypeReplaceRequest(SecretType secretType) throws SecretManagementException {
-
-        if (StringUtils.isEmpty(secretType.getName())) {
-            throw handleClientException(ERROR_CODE_SECRET_TYPE_NAME_REQUIRED, null);
-        }
-
-        if (!isSecretTypeExists(secretType.getName())) {
+    private void validateSecretDescription(String description)
+            throws SecretManagementClientException {
+        if (description != null && !isSecretDescriptionRegexValid(description)) {
```

Comment:
is discription madatory?

Reply:
No. This is the related regex, `"^.{0,1023}$"` which currently only checks for the length